dffml/df/types.py

```diff
@@ -361,6 +361,29 @@ def _fromdict(cls, **kwargs):
         return cls(**kwargs)
 
 
+@dataclass
+class Forward:
+    book: "Dict[str, List[Definitions]]" = None
+
+    def __post_init__(self):
+        if self.book is None:
+            self.book = {}
+        self._internal_book = []
+
+    def add(self, instance_name, definition_list):
+        self.book[instance_name] = definition_list
+        self._internal_book.extend(definition_list)
+
+    def get_instances_to_forward(self, definition):
+        if not definition in self._internal_book:
+            return []
+        return [
+            instance_name
+            for instance_name, definitions in self.book.items()
+            if definition in definitions
+        ]
```

Comment:
`export` and `_fromdict` in `DataFlow` need to be updated to correctly import and export this data structure.